src/output.cpp

```diff
@@ -1179,26 +1228,52 @@ int draw_item_info(WINDOW *win, const std::string sItemName,
             }
         }
     }
-    if( !buffer.str().empty() ) {
-        const auto b = without_border ? 1 : 2;
-        const auto width = getmaxx( win ) - b * 2;
-        fold_and_print( win, line_num, b, width, c_white, buffer.str() );
-    }
 
-    if (!without_border) {
-        draw_border(win);
-        wrefresh(win);
-    }
+    int ch;
+    do {
+        int iLines = 0;
+        if( !buffer.str().empty() ) {
+            const auto b = without_border ? 1 : 2;
+            const auto width = getmaxx( win ) - b * 2;
+            const auto height = getmaxy( win ) - 2;
+
+            const auto vFolded = foldstring(buffer.str(), width);
+            iLines = vFolded.size();
+
+            if (selected < 0) {
+                selected = 0;
+            } else if (iLines < height) {
+                selected = 0;
+            } else if (selected >= iLines - height) {
+                selected = iLines - height;
+            }
+
+            fold_and_print_from( win, line_num, b, width, selected, c_white, buffer.str() );
 
-    int ch = (int)' ';
-    if (!without_getch) {
-        ch = (int)getch();
-        if ( selected > 0 && ( ch == '\n' || ch == KEY_RIGHT ) && selected_ret != 0 ) {
-            ch = selected_ret;
-        } else if ( selected == KEY_LEFT ) {
-            ch = (int)' ';
+            draw_scrollbar(win, selected, height, iLines, 1, 0, c_white, false, iLines-height);
+        }
+
+        if (!without_border) {
+            draw_custom_border(win, buffer.str().empty());
+            wrefresh(win);
         }
-    }
+
+        ch = (int)' ';
+        if (!without_getch) {
+            ch = (int)getch();
+            if ( handle_scrolling && ch == KEY_PPAGE ) {
+                selected--;
+                werase(win);
+            } else if ( handle_scrolling && ch == KEY_NPAGE ) {
+                selected++;
+                werase(win);
+            } else if ( selected > 0 && ( ch == '\n' || ch == KEY_RIGHT ) && selected_ret != 0 ) {
+                ch = selected_ret;
+            } else if ( selected == KEY_LEFT ) {
+                ch = (int)' ';
+            }
+        }
+    } while (handle_scrolling && !without_getch && (ch == KEY_PPAGE || ch == KEY_NPAGE));
```

Comment:
The loop condition looks as if it is supposed to mean "(loop while) input from user should be handled here and input was handled here". This is a bit redundant. It might be better to implement leaving the loop via a break inside the input-handling:
```C++
while(true) {
    ... /* display, redraw etc. */
    if( without_getch ) {
        break;
    }
    ch = (int)getch(); 
    if( handle_scrolling && ch == KEY_PPAGE ) { 
        ... /* implicitly continues the loop */
    } else if( handle_scrolling && ch == KEY_NPAGE ) { 
        ... /* implicitly continues the loop */
    } else if ( selected > 0 && ( ch == '\n' || ch == KEY_RIGHT ) && selected_ret != 0 ) { 
        ch = selected_ret; 
        break;
    } else if ( selected == KEY_LEFT ) { 
        ch = (int)' '; 
        break;
    } else {
        break;
    }
}
```
That way you avoid differences in handling the input and in the loop condition.

It's easier to add/remove handlers for keys (e.g. add support for the POS1/HOME and END keys). Whether or not the new handlers should leave the loop is defined right in their if-block, not in the loop condition. Currently adding a new key handler (e.g. END key scrolls down to the last line) requires me to add a `else if` statement checking for it, but also adding something to the loop condition to prevent leaving the loop when that key has been entered.